File: my-folder/0216-combination-sum-iii/solution.cpp
class Solution {
public:
    void recursion(vector<vector<int>> &ans, vector<int> temp, int sum, int i, int k, int n){
        if (sum > n || temp.size() > k) return;
        if (sum == n && temp.size() == k) {
            ans.push_back(temp);
            return;
        }
        for (int j = i; j <= 9; j++) {
            temp.push_back(j);
            recursion(ans, temp, sum + j, j + 1, k, n);
            temp.pop_back(); 
        }
    }

    vector<vector<int>> combinationSum3(int k, int n) {
        vector<vector<int>> ans;
        recursion(ans, {}, 0, 1, k, n);
        return ans;
    }
};

File: my-folder/0216-combination-sum-iii/solution.java
class Solution {
    void getCombination(List<List<Integer>> ans,List<Integer> temp,int n,int k,int ind,int sum){
        if(temp.size()==k && sum==n){
            ans.add(new ArrayList<>(temp));
            return;
        }
        if( ind > 9 || temp.size()>k || sum>n) return;
        for(int i = ind;i<=9;i++){
            temp.add(i);
            getCombination(ans,temp,n,k,i+1,sum+i);
            temp.remove(temp.size()-1);
        }

    }
    public List<List<Integer>> combinationSum3(int k, int n) {
        List<List<Integer>> ans = new ArrayList<>();
        List<Integer> temp = new ArrayList<>();
        getCombination(ans,temp,n,k,1,0);
        return ans;
        
    }
}
